### ISSUE TITLE
Change coupling metric

### ISSUE THREAD

**Issue by ChristianHuehn** [MaibornWolff/metric-gardener#360]
File coupling metric is currently not further described on how and what it will count. This should be fixed according to this list:

- [ ] Coupling should be made clear that it is currently only a class coupling metric per file
- [ ] If a class uses any amount of methods from another class it's count as 1
- [ ] Fluent Interface or also called Method Chaining, should also be counted if possible. As the following example explains this pretty good:   
```const a = new A();
a.buildB().buildC().buildD().doStuff();

// kinda equivalent to
const a: A = new A();
const b: B = a.buildB();
const c: C = b.buildC();
const d: D = c.buildD();
d.doStuff();
```
This means that you need knowledge over the implementation of the other classes and therefor coupling them together.
- [ ] We count statics too

### Background information ###
It can be helpful to also check the visual studio implementation of it: https://learn.microsoft.com/en-us/visualstudio/code-quality/code-metrics-class-coupling?view=vs-2022

**Comment by BridgeAR** [comment]
We should ideally have different type of coupling metrics, not only a single one.

**Comment by ChristianHuehn** [comment]
Exactly, that is our plan 👍 . But first we want a basic version, that works and is tested. Before we move on adding new coupling metrics.
